packages/pipeline/src/scripts/pull_missing_blocks.ts

```diff
@@ -0,0 +1,83 @@
+// tslint:disable:no-console
+import { web3Factory } from '@0x/dev-utils';
+import * as Parallel from 'async-parallel';
+import R = require('ramda');
+import 'reflect-metadata';
+import { Connection, ConnectionOptions, createConnection, Repository } from 'typeorm';
+
+import { Web3Source } from '../data_sources/web3';
+import { Block } from '../entities';
+import * as ormConfig from '../ormconfig';
+import { parseBlock } from '../parsers/web3';
+import { handleError } from '../utils';
+
+// Number of blocks to save at once.
+const BATCH_SAVE_SIZE = 1000;
+// Maximum number of requests to send at once.
+const MAX_CONCURRENCY = 10;
+// Maximum number of blocks to query for at once. This is also the maximum
+// number of blocks we will hold in memory prior to being saved to the database.
+const MAX_BLOCKS_PER_QUERY = 1000;
+// Block number when the Exchange contract was deployed to mainnet.
+// TODO(albrow): De-dupe this constant.
+const EXCHANGE_START_BLOCK = 6271590;
+
+let connection: Connection;
+
+(async () => {
+    connection = await createConnection(ormConfig as ConnectionOptions);
+    const provider = web3Factory.getRpcProvider({
+        rpcUrl: `https://mainnet.infura.io/${process.env.INFURA_API_KEY}`,
```

Comment:
PSA, Infura is moving to a new API key type. We need to switch over our key if we haven't already.

Reply:
API key is totally optional in our code. At one point it looks like it might make a difference for rate-limiting but now I think it's not so clear. We'll need to do a more thorough analysis and if we determine it makes a difference I'll come back to this.

Reply:
Yep, sent you the email I got from Infura. Basically, starting Jan 23rd: 

```
Requests using Dashboard-generated Project IDs will be prioritized over legacy keys during periods of high traffic activity. Legacy keys may experience slower response times.
```

And then on March 27

```
Support will officially stop for all requests not sent using an Infura Dashboard-generated Project ID.
```